walkeeper/src/receive_wal.rs

```diff
@@ -142,6 +144,47 @@ pub struct ReceiveWalConn {
     pub conf: WalAcceptorConf,
 }
 
+///
+/// Periodically request pageserver to call back.
+/// If pageserver already has replication channel, it will just ignore this request
+///
+fn request_callback(conf: WalAcceptorConf, timelineid: ZTimelineId) {
+    let addr = conf.pageserver_addr.unwrap();
+    let ps_connstr = format!(
+        "host={} port={} dbname={} user={}",
+        addr.ip(),
+        addr.port(),
+        "no_db",
+        "no_user",
+    );
+    let callme = format!(
+        "callmemaybe {} host={} port={} options='-c ztimelineid={}'",
+        timelineid,
+        conf.listen_addr.ip(),
+        conf.listen_addr.port(),
+        timelineid
+    );
+    loop {
+        info!(
+            "requesting page server to connect to us: start {} {}",
+            ps_connstr, callme
+        );
+        if let Ok(mut client) = Client::connect(&ps_connstr, NoTls) {
+            if client.simple_query(&callme).is_err() {
+                error!("Failed to send callme request to pageserver");
```

Comment:
It would be nice to include the error in the report message. Same for connection attempt.